scripts/quickstart

```diff
@@ -1,94 +1,150 @@
 #!/bin/bash -e
 
-CMD="$(realpath "$0")"
-DIR="$(dirname "${CMD}")"
-BASE_DIR="$(dirname "${DIR}")"
-CONFIG_DIR="${BASE_DIR}/config"
-CERTS_DIR="${CONFIG_DIR}/certs"
-
-DOMAIN=openbalena.local
-
-usage() {
-  echo "usage: $0 [-h] [-p] [-d DOMAIN] -U EMAIL -P PASSWORD"
-  echo
-  echo "  -p           patch hosts - patch the host /etc/hosts file"
-  echo "  -d DOMAIN    the domain name this deployment will run as, eg. example.com. Default is 'openbalena.local'"
-  echo "  -U EMAIL     the email address of the superuser account, used to login to your install from the Balena CLI"
-  echo "  -P PASSWORD  the password to use for the superuser account."
-  echo
-}
+BLACK=`tput setaf 0`
+RED=`tput setaf 1`
+GREEN=`tput setaf 2`
+YELLOW=`tput setaf 3`
+BLUE=`tput setaf 4`
+MAGENTA=`tput setaf 5`
+CYAN=`tput setaf 6`
+WHITE=`tput setaf 7`
+
+BOLD=`tput bold`
+RESET=`tput sgr0`
+
+quickstart() {
+
+  # for macos machines, we need proper OpenSSL...
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+    echo -e "${BLUE}INFO: macOS detected. Bringing in OpenSSL if installed..."
+    export PATH="/usr/local/opt/openssl/bin:$PATH"
```

Comment:
I'd generally prefer things don't interfere with my PATH, but if it is necessary can we at least check that `/usr/local/opt/openssl/bin` exists first?

Reply:
I agree with Will here -- if we require OpenSSL on the host, then we should just check that the binary exists and that it of the right version and just bail out with a helpful message if not. In particular for macOS, it's trivial to install OpenSSH and update the PATH (`brew install` and `PATH=$(brew --prefix):${PATH} ./scripts/quickstart`).